File: split/printwritewhatsit.pas
procedure printwritewhatsit(s: strnumber; p: halfword);
begin
  printesc(s);
  if mem[p + 1].hh.lh < 16 then
    printint(mem[p + 1].hh.lh)
  else if mem[p + 1].hh.lh = 16 then
    printchar(42)
  else
    printchar(45);
end;

File: cpp/printwritewhatsit.cpp
#include "printwritewhatsit.h"
#include "printesc.h"
#include "printint.h"
#include "printchar.h"

void printwritewhatsit(strnumber s, halfword p)
{
	printesc(s);
	if (info(p+1) < 16)
		printint(info(p+1));
	else 
		if (info(p+1) == 16)
			printchar('*');
		else
			printchar('-');
}
